cmd/snapshotgc/snapshotgc.go

```diff
@@ -122,3 +123,46 @@ func getUnassociatedNSSnapshots(
 
 	return unAssociatedSnaps, nil
 }
+
+// Keep a certain amount of pr/non-pr snapshots
+func getSnapshotsForRemoval(
+	cl client.Client,
+	snapshots []applicationapiv1alpha1.Snapshot,
+	prSnapshotsToKeep int,
+	nonPrSnapshotsToKeep int,
+	logger logr.Logger,
+) []applicationapiv1alpha1.Snapshot {
+	sort.Slice(snapshots, func(i, j int) bool {
+		return snapshots[j].CreationTimestamp.Before(&snapshots[i].CreationTimestamp)
```

Comment:
I see the unittest is passing, but I don't understand this order

I'd expect `return timeI < timeJ` i.e. `snapshots[i].CreationTimestamp.Before(&snapshots[j].CreationTimestamp)`

As is done with users in example here https://pkg.go.dev/sort#Slice


What am I reading wrong here? :D

Reply:
When writing it, I was not aware that the function is expected to be a `less` function. Might be that the interface intention was that sorts will be implemented in ascending order, and then will be followed by a call to `Reverse()`. I'm not sure whether there's any issue with implementing it with a so-called `greater` function instead.

Reply:
aaah I got it now, plese make a comment there that we are explicitly sorting it in reverse order, so we are keeping the latest snapshots, for poor developers like me

Reply:
done